pkg/gateway/http.go

```diff
@@ -0,0 +1,53 @@
+package gateway
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/pires/go-proxyproto"
+
+	"github.com/Azure/ARO-RP/pkg/proxy"
+)
+
+func (g *gateway) handleConnect(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	conn, ok := ctx.Value(contextKeyConnection).(*proxyproto.Conn)
+	if !ok {
+		g.log.Error("invalid contextKeyConnection")
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	host, port, err := net.SplitHostPort(r.Host)
+	if err != nil {
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	isAllowed, err := g.isAllowed(conn, host)
+	if err != nil {
+		g.log.Error(err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	if !isAllowed || port != "443" {
+		http.Error(w, http.StatusText(http.StatusForbidden), http.StatusForbidden)
+		return
+	}
+
+	proxy.Proxy(g.log, w, r, SocketSize)
+}
+
+func (g *gateway) checkReady(w http.ResponseWriter, r *http.Request) {
```

Comment:
this behaviour isn't right.  How about something like:
* metric and alert around lastChangefeedTime
* metric and alert around dbtoken refresher
* neither of the above conditions should take the gateway out of readiness - i.e. regardless of db inaccessibility the gw should service new and existing connections from existing clusters